### DISCUSSION TITLE
Spread into parent

### DISCUSSION THREAD

**Discussion by KyleReczek-AbsorbLMS** [apple/pkl#579] [Ideas]
**My Use case:**

I need to mirror the following Yaml file.
```
Qa123:
  task: abc
Qa124:
  task: abc
```

I'd like to do something like the following 

```
test.pkl
...{
  for(env in environments) {
    [env]: new Task {
       task: abc
    }
  }
}
```

Would adding the ability to spread into the root object in a file like this be possible?

**Comment by holzensp** [comment]
There's no need to spread properties into the parent, simply because you want the output to be something else. This is precisely the scenario for exposing `output.value`. What you likely want is for `test.pkl` to have
```
tasks: Mapping<Environment, Task> = new {
  for (env in environments) {
    [env] {
      task = "abc"
    }
  }
}

output {
  value = tasks
}

**Reply by KyleReczek-AbsorbLMS** [reply]
Ah okay great! For some reason that hadn't occurred to me to spread it in the output.

Thanks for the speedy reply!

**Reply by holzensp** [reply]
(As OP, might I get you to mark this as answered?)
